src/main/java/games/strategy/triplea/ui/mapdata/DefaultColors.java

```diff
@@ -0,0 +1,25 @@
+package games.strategy.triplea.ui.mapdata;
+
+import java.awt.Color;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+final class DefaultColors {
+  private final List<Color> colors = new ArrayList<>(Arrays.asList(
+      Color.RED,
+      Color.MAGENTA,
+      Color.YELLOW,
+      Color.ORANGE,
+      Color.CYAN,
+      Color.GREEN,
+      Color.PINK,
+      Color.GRAY));
+
+  /**
+   * @throws IndexOutOfBoundsException If the available default colors have been exhausted.
+   */
+  Color nextColor() {
+    return colors.remove(0);
```

Comment:
This really looks like a normal iterator scheme to me.
I'd suggest making the list static and immutable again and simply creating a new iterator for each instance of this class.

Reply:
Btw. The shortest (but probably not most efficient way) of creating an iterator for an array would be:
```
Arrays.stream(colors).iterator();
```

Reply:
That's a good observation.

That will change the "out of colors" behavior to throw `NoSuchElementException` instead of `IndexOutOfBoundsException`.  I checked the callers of this method and verified no one is dependent on `IndexOutOfBoundsException`, so the change should be ok.

Also, since you noted that `DefaultColors` is really just an iterator, I think `NoSuchElementException` makes more sense even over `IllegalStateException`.  I will go ahead and make this change.